xbmc.py

```diff
@@ -746,9 +745,23 @@ def executebuiltin(function):
     See: http://kodi.wiki/view/List_of_Built_In_Functions
 
     """
+    import re
+    import subprocess
 
     KodiStub.print_line("Executebuiltin: {0}".format(function), color=Colors.Blue)
 
+    try:
+        command, params = re.search(r'([A-Za-z]+)\(([^\)]+)\)', function).groups()
+    except AttributeError:
+        raise ValueError('Could not process %s', function)
+
+    if command.lower() in ['runplugin', 'playmedia']:
+        addon, route = re.search(r'plugin://([^/]+)(.*)', params).groups()
+        addon_info = addoninfo.read_addon_xml(os.path.join(__add_on_info.kodi_home_path, 'addons', addon, 'addon.xml'))
+        pluginsource = addon_info.get('pluginsource')
+
+        subprocess.Popen(['python', os.path.join(__add_on_info.kodi_home_path, 'addons', addon, pluginsource), route])
```

Comment:
Is there any reason why we should actually execute the Python script? Would it not suffice to just check the path exists and see if it is a valid Python file? I used `py_compile.compile()` in some other script to do that.

Reply:
I actually need this to be executed for my use case. I'm using this to test IPTV Manager, and it needs to execute other add-ons so they can send back data. I have added two example add-ons in the test-home for this that do this.

I don't think this is a security problem, since the test-home isn't external but added to the sources.

Reply:
That is true, but if other add-ons use the `xbmc.executebuiltin(RunPlugin())` to call other add-ons, those add-ons will then also start and use Sakee, and might not designed for that. But perhaps that would be more an issue for those devs that decide to use it.

Reply:
True